test/rpc.spec.js

```diff
@@ -0,0 +1,59 @@
+'use strict'
+
+const expect = require('expect')
+const rewire = require('rewire')
+
+let Rpc = rewire('../server/lib/rpc')
+
+describe('[Rpc]', () => {
+  describe('(getInfo)', () => {
+    beforeEach(() => { // reset the rewired functions
+      Rpc = rewire('../server/lib/rpc')
+    })
+    it('should run getinfo and fail the rpc call', (done) => {
+      Rpc.navClient = {
+        getInfo: () => { return Promise.reject({ code: -17 }) },
+      }
+      const res = {
+        send: (response) => {
+          const jsonResponse = JSON.parse(response)
+          expect(jsonResponse.type).toBe('FAIL')
+          expect(jsonResponse.code).toBe('RPC_002')
+          expect(jsonResponse.error.code).toBe(-17)
```

Comment:
Could test for a specific status code, at the moment only 200s are being thrown but that could always change in the future. For example we may eventually want to throw 50x codes when something screws up on the server side.